app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/signIn/forgotpassword/ForgotPasswordViewModel.kt

```diff
@@ -0,0 +1,162 @@
+
+    private var _code = MutableStateFlow("")
+    val code = _code.asStateFlow()
+
+    private var _hasEmailError = MutableStateFlow(false)
+    val hasEmailError = _hasEmailError.asStateFlow()
+
+    private var _getCodeErrorState = MutableStateFlow(false)
+    val getCodeErrorState = _getCodeErrorState.asStateFlow()
+
+    private var _resetPasswordErrorState = MutableStateFlow(false)
+    val resetPasswordErrorState = _resetPasswordErrorState.asStateFlow()
+
+    private var _resetPasswordSuccess = MutableStateFlow(false)
+    val resetPasswordSuccess = _resetPasswordSuccess.asStateFlow()
+
+    private var _resetPasswordErrorMessage = MutableStateFlow("")
+    val resetPasswordErrorMessage = _resetPasswordErrorMessage.asStateFlow()
+
+    private var _hasRepeatPasswordError = MutableStateFlow(false)
+    val hasRepeatPasswordError = _hasRepeatPasswordError.asStateFlow()
+
+
+    private var _getCodeErrorMessage = MutableStateFlow("")
+    val getCodeErrorMessage = _getCodeErrorMessage.asStateFlow()
+
+    private var _hash = MutableStateFlow("")
+
+    private var _hasGetCodeSuccess = MutableStateFlow(false)
+    val hasGetCodeSuccess = _hasGetCodeSuccess.asStateFlow()
+
+
+    val enableDoneButton = combine(
+        _hasPasswordError,
+        _hasRepeatPasswordError
+    ) { passwordError, repeatPasswordError ->
+        _password.value.isNotEmpty()
+                && _repeatPassword.value.isNotEmpty()
+                && !passwordError && !repeatPasswordError
+    }.stateIn(
+        viewModelScope, SharingStarted.WhileSubscribed(), false
+    )
+
+    fun getCode() {
+        viewModelScope.launch {
+            forgotPasswordRepository.resetPasswordEmail(_email.value).onEach {
+                it.onSuccess { getCodeResponse ->
+                    _hash.value = getCodeResponse?.hash.toString()
+                    _hasGetCodeSuccess.value = true
+                }
+                it.onFailure { error ->
+                    _getCodeErrorMessage.value = error.message ?: "Error"
+                    _hasGetCodeSuccess.value = false
+                }
+            }.catch {
+                _getCodeErrorState.value = true
+            }.flowOn(Dispatchers.Main).collect()
+        }
+    }
+
+    fun resetPassword() {
+        viewModelScope.launch(Dispatchers.IO) {
+            forgotPasswordRepository.forgotPassword(
+                ForgotPasswordEntity(
+                    hash = _code.value,
+                    email = _email.value,
+                    password = _password.value
+                )
+            ).onEach {
+                it.onSuccess { _ ->
+                    _resetPasswordSuccess.value = true
+                }.onFailure { error ->
+                    _resetPasswordErrorMessage.value = error.message ?: "Error"
+                }
+                clearForm()
+            }.catch {
+                _resetPasswordErrorState.value = true
+            }.flowOn(Dispatchers.Main)
```

Comment:
This `Dispatcher.Main` contradicts with `viewModelScope.launch(Dispatchers.IO)`. I think we don't need `flowOn(Dispatchers.Main)` here.